session.go

```diff
@@ -421,6 +629,54 @@ func SendEvent(eventType string, eventData map[string]interface{}, qos QOSLevel)
 	return <-ctrl.response
 }
 
+func SetKeyValue(key string, value map[string]interface{}) error {
+	ctrl := &sessCtrlSendKeyValue{
+		keyValue: &KeyValue{Action: "set", Key: key, Value: value},
+		response: make(chan error, 1),
+	}
+
+	sessCtrl.sendKeyValue <- ctrl
+	return <-ctrl.response
+}
+
+func GetKeyValue(key string) (*KeyValue, bool) {
```

Comment:
I think still GetKeyValue/GetAllKeyValues may have bugs. Because it cannot handle while idle/reconnecting.

To simplify, should I use reader/writer lock simply to get the value from kvStore?